code-challenges.test.js

```diff
@@ -22,30 +22,53 @@ var secretCodeWord1 = "lackadaisical"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
-
-
+describe ("when 'enigmaMachine' is called", () => {
+    let secretNaziWord1 = "lackadaisical"
+    let code1 = "l4ck4d41s1c4l"
+    let secretNaziWord2 = "gobbledygook"
+    let code2 = "g0bbl3dyg00k"
+    it ("replaces certain letters", () => {
+        expect (enigmaMachine(secretNaziWord1)).toEqual(code1)
+    })
+    it ("replaces certain letters", () => {
+        expect (enigmaMachine(secretNaziWord2)).toEqual(code2)
+    })
+})
 
 
 // b) Create the function that makes the test pass.
 
-
-
+const enigmaMachine = (string) => {
+    let coded = string.replace(/e/ig,'3').replace(/a/ig,'4').replace(/i/g,'1').replace(/o/g,'0')
```

Comment:
good logic. Consider a way to organize the mapping from letter to number without having to iterate on the string multiple times.